### ISSUE TITLE
Adding hyperparameter tuning for different model to  improve accuracy

### ISSUE THREAD

**Issue by Prasadayus** [Recode-Hive/Stackoverflow-Analysis#97]
hi @sanjay-kv 
we can add hyperparameter tuning to improve accuracy for this ,for this we can use optuna also.We can add Xgboost model also.
I will be very glad to work on this issue.
